migration/init_test.go

```diff
@@ -0,0 +1,50 @@
+package migration
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/store"
+)
+
+func TestGenesisInitializeSchemaVersions(t *testing.T) {
+	const genesis = `
+	{
+		"conf": {
```

Comment:
What happens when you load this?

`gconf.Load(kv, "migration", &Configuration{})`

I assume Configuration will have no metadata or schema, which could cause an error.
How to handle versioning for Configuration?

We could force it to be included :(

Or we could have gconf depend on migration.
Require "initialize_schema" to be processed first.
And then `InitConfig` auto-sets the schema of the configuration to the active schema value for that package stored in migrations?

But maybe that is just too much complexity to save one line in the genesis file :thinking:

Reply:
I don't know :see_no_evil:  I never thought about it. I am already implicitly registering schema for the `migration` package. I can do the same for other weave components that are not opional, for example `gconf` https://github.com/iov-one/weave/pull/556/files#diff-5a28220c3fc3e4a1cf807b52300f8840R36

Reply:
The configuration object is unknown to gconf. 
But if you do add the Metadata to it, it needs to be validated.

Or remove Metadata from Configuration now. And add it in another PR when we can think this through better. It is not blocking, but the feature is broken now

Reply:
Metadata removed in https://github.com/iov-one/weave/pull/556/commits/cf77ccfaa5ca7800e0a0ee43c3d3f646755e7276